yt_dlp/extractor/niconico.py

```diff
@@ -367,229 +388,120 @@ def parse_format_id(id_code):
-
-        duration = (parse_duration(
-            get_video_info_web('length')
-            or self._html_search_meta(
-                'video:duration', webpage, 'video duration', default=None))
-            or video_detail.get('length')
-            or get_video_info_web('duration'))
-
-        webpage_url = get_video_info_web('watch_url') or url
-
-        # for channel movie and community movie
-        channel_id = try_get(
-            api_data,
-            (lambda x: x['channel']['globalId'],
-             lambda x: x['community']['globalId']))
-        channel = try_get(
-            api_data,
-            (lambda x: x['channel']['name'],
-             lambda x: x['community']['name']))
-
-        # Note: cannot use api_data.get('owner', {}) because owner may be set to "null"
-        # in the JSON, which will cause None to be returned instead of {}.
-        owner = try_get(api_data, lambda x: x.get('owner'), dict) or {}
-        uploader_id = str_or_none(
-            get_video_info_web(['ch_id', 'user_id'])
-            or owner.get('id')
-            or channel_id
-        )
-        uploader = (
-            get_video_info_web(['ch_name', 'user_nickname'])
-            or owner.get('nickname')
-            or channel
-        )
+                timestamp = parse_iso8601(match)
+
+        comment_count = traverse_obj(
+            api_data, ('video', 'count', 'comment'), expected_type=int)
+
+        duration = (
+            parse_duration(self._html_search_meta('video:duration', webpage, 'video duration', default=None))
+            or get_video_info('duration'))
+
+        if url.startswith('http://') or url.startswith('https://'):
+            webpage_url = url
+        else:
+            webpage_url = 'https://www.nicovideo.jp/watch/%s' % video_id
+
+        uploader_id = traverse_obj(api_data, ('owner', 'id'))
+        uploader = traverse_obj(api_data, ('owner', 'nickname'))
+
+        channel_id = traverse_obj(api_data, ('channel', 'id'), ('community', 'id'))
+        channel = traverse_obj(api_data, ('channel', 'name'), ('community', 'name'))
+
+        tags = None
+        if webpage:
+            # use og:video:tag (not logged in)
+            og_video_tags = re.finditer(r'<meta\s+property="og:video:tag"\s*content="(.*?)">', webpage)
+            tags = list(filter(None, (clean_html(x.group(1)) for x in og_video_tags)))
+            if not tags:
+                # use keywords and split with comma (not logged in)
+                kwds = self._html_search_meta('keywords', webpage, default=None)
+                if kwds:
+                    tags = [x for x in kwds.split(',') if x]
+        if not tags:
+            # find in json (logged in)
+            tags = traverse_obj(api_data, ('tag', 'items', ..., 'name'))
+
+        genre = traverse_obj(api_data, ('genre', 'label'), ('genre', 'key'))
+
+        subtitles = None
+        if self.get_param('getcomments', False) or self.get_param('writesubtitles', False):
```

Comment:
Either treat it as subs, or treat it as comment, not both

Reply:
Should be comment then. I'll fix it

Reply:
*subtitles